src/umpire/Umpire.hpp

```diff
@@ -189,6 +192,28 @@ MPI_Comm get_communicator_for_allocator(Allocator a, MPI_Comm comm);
 void register_external_allocation(void* ptr, util::AllocationRecord record);
 util::AllocationRecord deregister_external_allocation(void* ptr);
 
+/*!
+ * \brief Returns the Camp resource associated with a particular allocation
+ *
+ * \param Umpire allocator which was used to allocate the data
+ * \param Pointer which was used for the allocation
+ *
+ * \return Camp resource associated with the allocation *assuming the Allocator
+ * passed in is a ResourceAwarePool strategy and the allocation is either used or pending*
+ */
+Resource getResource(Allocator a, void* ptr);
+
+/*!
+ * \brief Returns the current number of pending chunks associated with the pool. The
+ * pending chunks are those that have been scheduled to deallocate but may or may not
+ * have actually been deallocated yet.
+ *
+ * \param Umpire allocator with ResourceAwarePool strategy
+ *
+ * \return Number of pending chunks in the ResourceAwarePool
+ */
+std::size_t getPendingSize(Allocator a);
```

Comment:
Is this the size of pending memory or the number of chunks containing pending memory?

Reply:
the number of chunks - perhaps the function name could be made clearer...